tests/dynamic_batch_sizes.rs

```diff
@@ -0,0 +1,195 @@
+        "--database",
+        &db_url,
+        "--database-max-connections",
+        "1",
+        "--tree-depth",
+        &format!("{tree_depth}"),
+        "--prover-urls",
+        &prover_mock.arg_string(),
+        "--batch-timeout-seconds",
+        "3",
+        "--dense-tree-prefix-depth",
+        "10",
+        "--tree-gc-threshold",
+        "1",
+    ])
+    .context("Failed to create options")?;
+
+    options.server.server = Url::parse("http://127.0.0.1:0/").expect("Failed to parse URL");
+
+    options.app.contracts.identity_manager_address = mock_chain.identity_manager.address();
+    options.app.ethereum.read_options.confirmation_blocks_delay = 2;
+    options.app.ethereum.read_options.ethereum_provider =
+        Url::parse(&mock_chain.anvil.endpoint()).expect("Failed to parse Anvil url");
+
+    options.app.ethereum.write_options.signing_key = mock_chain.private_key;
+
+    let (app, local_addr) = spawn_app(options.clone())
+        .await
+        .expect("Failed to spawn app.");
+
+    let test_identities = generate_test_identities(batch_size * 5);
+    let identities_ref: Vec<Field> = test_identities
+        .iter()
+        .map(|i| Hash::from_str_radix(i, 16).unwrap())
+        .collect();
+
+    let uri = "http://".to_owned() + &local_addr.to_string();
+    let client = Client::new();
+
+    // Insert enough identities to trigger an batch to be sent to the blockchain
+    // based on the current batch size of 3.
+    test_insert_identity(&uri, &client, &mut ref_tree, &identities_ref, 0).await;
+    test_insert_identity(&uri, &client, &mut ref_tree, &identities_ref, 1).await;
+    test_insert_identity(&uri, &client, &mut ref_tree, &identities_ref, 2).await;
+
+    // Check that we can get their inclusion proofs back.
+    test_inclusion_proof(
+        &uri,
+        &client,
+        0,
+        &ref_tree,
+        &Hash::from_str_radix(&test_identities[0], 16)
+            .expect("Failed to parse Hash from test leaf 0"),
+        false,
+    )
+    .await;
+    test_inclusion_proof(
+        &uri,
+        &client,
+        1,
+        &ref_tree,
+        &Hash::from_str_radix(&test_identities[1], 16)
+            .expect("Failed to parse Hash from test leaf 0"),
+        false,
+    )
+    .await;
+    test_inclusion_proof(
+        &uri,
+        &client,
+        2,
+        &ref_tree,
+        &Hash::from_str_radix(&test_identities[2], 16)
+            .expect("Failed to parse Hash from test leaf 0"),
+        false,
+    )
+    .await;
+
+    // Add a new prover for batch sizes of two.
+    let second_prover = spawn_mock_prover(second_batch_size).await?;
+    test_add_batch_size(&uri, second_prover.url(), second_batch_size as u64, &client)
+        .await
+        .expect("Failed to add batch size.");
+
+    // Insert enough identities to trigger the lower batch size.
+    test_insert_identity(&uri, &client, &mut ref_tree, &identities_ref, 3).await;
```

Comment:
Could we maybe also make the larger prover unavaiable here?

With

```
prover_mock.set_availability(false).await;
```

Just to be sure that we're using the correct prover in this case.